pkg/ddc/base/operation.go

```diff
@@ -100,15 +102,9 @@ func (t *TemplateEngine) reconcilePending(ctx cruntime.ReconcileRequestContext,
 	opStatus *datav1alpha1.OperationStatus, operation dataoperation.OperationInterface) (ctrl.Result, error) {
 	log := ctx.Log.WithName("reconcilePending")
 
-	// 1. lock the dataset
-	err := SetDataOperationInTargetDataset(ctx, object, operation, t)
-	if err != nil {
```

Comment:
`SetDataOperationInTargetDataset` used for:
- lock the target dataset
- set the target dataset status ( status changed to `DataMigrating`  for datamigrate)

Does the second scenario still need?

Reply:
Thanks for your advices! You are right. I del lock in this function. Please have a look.

BTW, concurrent datamigrates should check if there are other migrates running and deal with dataset status (in RemoveTargetDatasetStatusInProgress).  What do you think?